### PR TITLE
PYTHON-4134: Update LangChain test to run langchain-mongodb partner package rather than community

### PR DESCRIPTION
**Pull request by Jibola**
## Summary
Our LangChain tests have evolved immensely. The largest change stems from test running in their own partner library langchain-mongodb. As a result, we are dropping our support/maintenance of the langchain-community package version and are wholesale supporting our new langchain-mongodb partner package. **The goal of this PR is to now run the langchain-mongodb test suite**

- [x] Blocked until #11 #12 are merged
- [x] Blocked until #21 is merged

### CODE REVIEW

File: langchain-python/run.sh [hunk 1]
```diff
@@ -7,21 +7,18 @@ set -x
...
 source ./bin/activate
 
 pip install poetry
-pip install filelock
-pip install msgpack
-pip install requests_toolbelt
-pip install motor
 
 poetry install --with test
 # Workaround provided in https://github.com/langchain-ai/langchain/issues/12237
```
Comment (blink1073) [inline langchain-python/run.sh:18]: Should we remove this block now that we don't need a workaround?

**Review by blink1073 — CHANGES_REQUESTED** [review]

File: langchain-python/run.sh (hunk 1 above)
Comment (caseyclements) [reply langchain-python/run.sh:18]: According to the Poetry docs, "...the isolated virtual environment in which poetry is installed should not be activated for running poetry commands." 

I'm going through this stuff now. What they recommend is more like this

```
PYTHON_BINARY -m venv $VENV_PATH
$VENV_PATH/bin/pip install -U pip setuptools
$VENV_PATH/bin/pip install poetry 
$VENV_PATH/bin/pip install poetry install --with test
```

Comment (caseyclements) [reply langchain-python/run.sh:18]: The following appears to work well for me.

```
PYTHON_BINARY=$(find_python3)
$PYTHON_BINARY -c "import sys; print(f'python version found: {sys.version_info}')"

$PYTHON_BINARY -m pip install poetry
poetry env use python3.10
source $(poetry env info --path)/bin/activate
poetry install --with dev

# Stored in evergreen VARIABLES
# source sandbox/setenv.sh. <== Disregard. This was from running locally. I'm not set up with CI approach just yet.

poetry run pytest tests/datastore/providers/mongodb_atlas/
```

Comment (caseyclements) [reply langchain-python/run.sh:18]: How do we actually set the python version?

Comment (Jibola) [reply langchain-python/run.sh:18]: The venv python version? In my version, I'm still keeping to the poetry heuristic. I'm just walling everything off with a venv.

Comment (caseyclements) [reply langchain-python/run.sh]: Hey. This was an OLD comment, I think. The code in `find_python3` is pretty simple.

**Review by caseyclements — CHANGES_REQUESTED** [review]
Update poetry usage to avoid venv.

File: langchain-python/run.sh [hunk 2]
```diff
@@ -7,21 +7,15 @@ set -x
 
 PYTHON_BINARY=$(find_python3)
 
-cd libs/langchain
+cd libs/partners/mongodb
 
 $PYTHON_BINARY -m venv .
```
[the remark this answers is not in the document]
Comment (Jibola) [reply langchain-python/run.sh:12]: Ah yeah, I think the _one_ thing about that is the evergreen urges not ["writing outside of the directory"](https://docs.devprod.prod.corp.mongodb.com/evergreen/Project-Configuration/Best-Practices/#:~:text=Please%20don%27t%20write%20outside%20this%20directory%2C%20as%20Evergreen%20won%27t%20be%20able%20to%20delete%20the%20data%20your%20task%20has%20written.) which is what I interpreted a pip install with the binary to be under that general umbrella.

[the remark this answers is not in the document]
Comment (caseyclements) [reply langchain-python/run.sh:12]: Ah. You're right. Using a venv will keep all in one place. Hmmm. Maybe I should update the others...

**Review by caseyclements — APPROVED** [review]
LGTM

**Review by caseyclements — APPROVED** [review]
Given merge of #21 , good to go.

**Pull request merged**